Azure Monitor: Logs Query Builder - Data Transformations

alyssajoyner opened the pull request:
This PR is part of the [larger PR effort](https://github.com/grafana/grafana/pull/99055) to implement a Builder Mode for Azure Monitor. 

This adds the following functionality: 

- [x] The ability to apply Filtering, Aggregations, and GroupBy.
- [x] Refactors the way KQL queries are built with the introduction of `KustoExpressionQueryParser.tsx`

Closes #99586

alyssajoyner committed:
Add filtering

yuri-tceretian committed:
Alerting: Rule version history API (#99041)

* implement store method to read rule versions

* implement request handler

* declare a new endpoint

* fix fake to return correct response

* add tests

* add integration tests

* rename history to versions

* apply diff from swagger CI step

Signed-off-by: Yuri Tseretyan <yuriy.tseretyan@grafana.com>

---------

Signed-off-by: Yuri Tseretyan <yuriy.tseretyan@grafana.com>

alyssajoyner committed:
Merge main

NWRichmond committed:
ExploreMetrics: Add toggle to enable routing to externalized Explore Metrics app plugin (#99481)

* feat: toggle for explore metrics app plugin

* chore: put app navigation settings behind feature toggle

* chore: require restart

* feat: context-driven applinks toggling

* fix: toggle ownership

* docs: note restart requirement

* refactor: prefer global feature toggle check

* chore: undo rm newlines

* fix: minimize changes

* fix: app id, `chromeless`, sidebar ordering

scottlepp committed:
[search] fix folder sync (#99913)

[search] fix folder sync

ppcano committed:
Alerting docs: update screenshot (modify NoData/Error state) (#99996)

baldm0mma committed:
Chore: Update alpine build docker image (#99998)

baldm0mma/ update alpine build image

baldm0mma committed:
Chore: Manually update versions and update `grafana-test-datasource` to "workspace:*" (#99999)

* baldm0mma/ manually update deps and run yarn

* baldm0mma/ update version from workspace to actual version

* baldm0mma/ revert some workspaces

* baldm0mma/ update workspace

* baldm0mma/ update deps

stephaniehingtgen committed:
Folders: Validator: add temporary fix (#100019)

alyssajoyner committed:
Add filtering and clean up code preview

alyssajoyner committed:
Lint

stephaniehingtgen committed:
K8s: Fix dashboard deletion in folders (#100023)

Clarity-89 committed:
RTK APIs: Support multiple source files (#100029)

ashharrison90 committed:
API client generation: Add `verify-api-clients` step to drone (#99987)

* add verify-api-clients step to drone

* change to check the verify step fails

* ignore conf dir

* rerun make drone

* undo change to make step fail

yincongcyincong committed:
Dashboard: Playlist - Fix issue with back button (#99401)

* Playlist: Goback not work
* add unit test
* format code

axelavargas committed:
Dashboards: SchemaV2 - Fix saving dashboards on folder (#100037)

* Remove folder annotations not relevant to v2 api
* Add unit test for folder annotations

bfmatei committed:
Bump Scenes to 5.41.2 (#100036)

brendamuir committed:
Adds johnny as codeowner for alerting docs (#99877)

* Adds johnny as codeowner for alerting docs

* changes maintainer

bfmatei committed:
DashboardList: Throttle the re-renders (#99982)

ivanahuckova committed:
Report grpc canceled timeouts as cancelled status and not error (#99981)

AgnesToulet committed:
Share: Fix short links when root_url is different from the browser URL (#99950)

Share: Fix short links when appUrl is misconfigured

kalleep committed:
Auth: Update authlib (#100041)

Bump authlib

mdvictor committed:
DynamicDashboards: Fix scroll positioning when returning to dashboard (#100047)

* fix scroll positioning in new layouts editor

* refactor

imatwawana committed:
Docs: fix link text (#99932)

toddtreece committed:
Dashboards: Fix schemaVersion test order (#100016)

beejeebus committed:
Add errorsource to InfluxDB datasource plugin fixes #1072 (#99900)

This PR adds `backend.ErrorSourceDownstream` values to all `backend.DataResponse`
values where it's certain that the error wasn't the result of the
InfluxDB datasource plugin.

sknaumov committed:
Dashboards: Allow custom quick time ranges specified in dashboard model (#93724)

* TimeRangePicker: allow to customize quick ranges per dashboard

* TimeRangePicker: show selected custom time range using its name

* rangeutil: add tests for describeTextRange + quickRanges

* Fix up tests, and add an extra case for hidden time ranges

* Don't construct an object to find options, add findRangeInOptions util

* fix type errors detected by TypeScript

---------

Co-authored-by: joshhunt <josh@trtr.co>

mdvictor committed:
DynamicDashboards: Open edit overlay on selection if pane is collapsed (#99885)

* open edit pane on selection

* stuff

* use drawer instead for override panel options

* reverts

* closing edit pane while selection exists should also clear selection

* set width

tomratcliffe committed:
Alerting: Update design of rule details tab and add `updated by` (#99895)

kalleep committed:
Auth: Remove feature toggle `authAPIAccessTokenAuth` (#100055)

Remove feature toggle

clord committed:
Plugin Metrics: Eliminate data race in plugin metrics middleware (#99396)

fix: eliminate data race in plugin metrics middleware

A data race was detected when multiple goroutines accessed the `MetricsMiddleware`
simultaneously. The race occurred because a single `MetricsMiddleware` instance
was being shared across goroutines while its `BaseHandler` field was being
modified during middleware chain setup.

Fix by creating a new `MetricsMiddleware` instance for each middleware chain,
while safely sharing the thread-safe Prometheus metrics and plugin registry.
This maintains proper metrics collection while eliminating the mutable shared
state that caused the race condition.

Original error was detected here:

```
WARNING: DATA RACE
Read at 0x00c0039c0790 by goroutine 4486:
  github.com/grafana/grafana-plugin-sdk-go/backend.(*ErrorSourceMiddleware).CallResource()
      /Users/clord/src/grafana/irm-devstack/.devenv/state/go/pkg/mod/github.com/grafana/grafana-plugin-sdk-go@v0.261.0/backend/error_source_middleware.go:93 +0x40
  github.com/grafana/grafana-plugin-sdk-go/backend.BaseHandler.CallResource()
  ...

```

toddtreece committed:
Dashboards: Apply schemaVersion migration in v2 conversion (#99973)

svennergr committed:
Explore: Fix casing for `exploreHideLogsDownload` setting (#100081)

matyax committed:
Logs Panel: Base elements for the new visualization (#99084)

* Create base components

* Create measurement service

* Add container for list

* Use measurement to render virtualized log lines

* Match rendered styles in 2d context for measuring

* Improve virtualization initialization and handle resize

* Introduce log line processing

* Virtualization: fix measurement of lines with line endings

* Virtualization: include scrollbar width in calculation

* Remove logs

* Virtualization: optimize text measurement

* Add support for forceEscape

* Log line: properly style wrapped/unwrapped lines

* Virtualization: handle possible overflows

* Improve overflow handling

* LogList: remove scroll position ref

* Remove logs

* Remove log

* Add top/bottom navigation buttons

* Add timestamp to pre-processing

* Add showtime support

* Fix imports

* Chore: simplify dedup

* Show level

* Refactor measurement and measure level and timestamp

* Virtualization: skip unnecessary measurements

* Improve measurements to minimize overflow chance

* Introduce logline colors

* Update palette

* Remove pretiffying

* Add comment

* Remove unused variable

* Add color for info level

* Fix dependencies

* Refactor overflow to account for smaller estimations

* Debounce resizing

* Fix imports

* Further optimize height calculation

* Remove outline

* Unused import

* Use less under/overflow method

* Respond to height changes

* Refactor size adjustment to account for layout changes

* Add Logs Panel support

* Add margin bottom to log lines

* Remove unused option

* LogList: container div should never be null

Bad API design

* Log List: make app not undefined and update containerElement usages

* New Logs Panel: Create as new visualization (#99427)

* Logs Panel: clean up old panel

* Logs Panel New: create as new visualization

* Plugin: mark as alpha

* Logs panel new: hold container in a state variable

* Logs panel: fix no data state

* Create newLogsPanel feature flag

* Logs: use new feature flag

* Prettier

* Add new panel to code owners

* Logs Navigation: add translations

* Address betterer issues

* Fix import

* Extract translations

* Update virtualization.ts

* Virtualization: add DOM fallback for text measurement

* Run gen-cue

* plugins_integration_test: add logs-new to expected plugins

nmarrs committed:
Image Renderer: Add support for SSL in plugin mode (#98009)

yuri-tceretian committed:
Alerting: set updated_by for system owned operations (#100068)

owensmallwood committed:
Unified Storage: Updates index latency logging (#100085)

updates index latency logging

bohandley committed:
Explore metrics: Fix otel bug (#100092)

toddtreece committed:
Dashboards: Fix title conversions (#100084)

dmihai committed:
API keys: Migrate API keys to service accounts at startup (#96924)

* migrate API keys to SA at startup

* send metrics with api key migration stats

* address feedback

* run API keys migration in a server lock

* update logging

teodosii committed:
Alerting: Update IRM copies in Configuration Tracker (#100069)

* updated alerting configuration tracker with new IRM copies

* Configuration tracker steps

* copy

* CI trigger

---------

Co-authored-by: Sonia Aguilar <soniaaguilarpeiron@gmail.com>

bfmatei committed:
Dashboards: Refactor types for dynamic dashboards (#100064)

NWRichmond committed:
ExploreMetrics: Use redirect to route traffic to metrics drilldown app (#100098)

* fix: drilldown app redirect

* chore: make copilot review happy, I guess 🤷

bfmatei committed:
Dashboards: Use i18n for texts in dynamic dashboards (#100108)

IevaVasiljeva committed:
RBAC: Don't check folder access if `annotationPermissionUpdate` FT is enabled (#99717)

don't check folder access if annotation permission update is enabled

dana-axinte committed:
CloudMigrations: Update link to cloud migration assistant page  (#99915)

* change link

* consolidate banners

* remove banner on plugins

ivanortegaalba committed:
Dashboard V0->V1 Migration: Schema migration v39 (#99631)

jtvdez committed:
Docs: Adding info on decrypting encrypted certificates (#100020)

andresmgot committed:
Advisor: Fix Healtcheck check (#100115)

kalleep committed:
Authz: Check for parent uid instead of id (#100121)

* Check for parent uid instead of id

kalleep committed:
Auth: Fix function name (#100122)

Fix spelling

joshhunt committed:
FolderPicker: Make lazy in prep for exposing publicly (#100118)

* Make lazy NestedFolderPicker

* Change permission prop to use string union instead of enum

* reword comment

JacobsonMT committed:
Alerting: Add EmbeddedContents as alternative embedding in smtp (#99937)

Adds support for embedding []byte in SmtpClient instead of filenames. This is backwards compatible as it uses a new field EmbeddedContents to add an alternative to the existing EmbeddedFiles which takes filenames.

ArturWierzbicki committed:
Dashboards: Update `POST /api/dashboards/db` docs (#99363)

update dashboard update docs

imatwawana committed:
Docs: move missing ref URI to correct page (#100131)

colin-stuart committed:
Docs: Add docs for Passwordless Authentication Using Magic Links (#96877)

* Docs: Add docs for Passwordless Authentication Using Magic Links

* Update docs/sources/setup-grafana/configure-security/configure-authentication/passwordless/index.md

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

* Update docs/sources/setup-grafana/configure-security/configure-authentication/passwordless/index.md

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

* Update docs/sources/setup-grafana/configure-security/configure-authentication/passwordless/index.md

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

* match Writer's Toolkit style

* Update docs/sources/setup-grafana/configure-security/configure-authentication/passwordless/index.md

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

---------

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>
Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

colin-stuart committed:
Auth: Add IP address login attempt validation (#98123)

* Auth: Add IP address login attempt validation

* LoginAttempt struct IpAddress field must be camelCase to match db ip_address column

* add setting DisableIPAddressLoginProtection

* lint

* add DisableIPAddressLoginProtection setting to tests

* add request object to authenticate password test

* nit suggestions & rename tests

* add login attempt on failed password authentication

* dont need to reset login attempts if successful

* don't change error message

* revert go.work.sum

* Update pkg/services/authn/clients/password.go

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

---------

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

kminehart committed:
CI: Backport to release branches (#100067)

* update backport and release comms

* Backport to release branches and change docs source branch for publishing

* Add new workflows to CODEOWNERS

* Re-add removed line oops

* backport-testing -> grafana

* checkout grafana repo in backport action, reference repo / branch in reusable action

* generate -> create

yuri-tceretian committed:
Alerting: Remove ID and OrgID from hash calculation (#100140)

kevinwcyu committed:
CI: Commit package.json changes in e2e/test-plugins when bumping versions (#97506)

kminehart committed:
CI: remove unnecessary line in backport.yml (#100144)

remove bad line

JohnnyQQQQ committed:
feat(unified-storage): use continue token when building list (#100143)

toddtreece committed:
Dashboard: Add v1alpha1 test coverage (#100149)

bohandley committed:
Explore metrics: Add option to use regex (#100146)

add option to use regex

yuri-tceretian committed:
Alerting: Remove id and org_id from grafana alert rule API model (#100139)

alyssajoyner committed:
Add GroupBy

stephaniehingtgen committed:
FindDashboards: filter by dashboard type (#100160)

bossinc commented:
This is being committed into `[alyssa/azure-logs-simple-mode](https://github.com/grafana/grafana/tree/alyssa/azure-logs-simple-mode)` instead of main. Is that expected?

alyssajoyner committed:
refactor kql query creation

andresmgot committed:
Advisor: Refactor check interfaces (#100043)

ashharrison90 committed:
Storybook: Support an arbitrary number of themes (#100111)

* support more themes in storybook

* default to dark theme

* fix type error

* change theme in docs container

* add TODO

* only show extra themes in development mode

* add comment

leeoniya committed:
ComboBox: Fall back to substring matching for symbols/operators (#100148)

torkelo committed:
Dashboard: Various fixes to new layouts (#100107)

* Dashboard: Various fixes to new layouts

* review fixes

* Fix

* Update

* Fix test

kalleep committed:
Dashboards+Folders: Ensure the service identity is used for resolvers (#100128)

* Dashboards+Folders: Ensure the service identity is used for dashboard and folder resolvers

* Add convinient function to call closure with service context

ashharrison90 committed:
Themes: Add new theme definitions behind feature toggle (#100129)

* create new toggle

* add survey link behind feature toggle

* fix translations

* better theme structure

* add all the themes back

* update matrix

* fix mars contrast

* fix color contrast probs with tron

* fix a11y issues with synthwave/victorian themes

* fix aubergine/zen

* rename green + gold

* rename to space

* rename, only enable 4 for grafanacon

* add survey link

* fix info color in sapphiredusk

* handle extra themes in storybook

macabu committed:
CloudMigrations: save snapshot of alert rule groups (#100109)

yesoreyeram committed:
[analytics] added plugin version to grafana_ds_test_datasource_clicked event (#100168)

matyax committed:
Loki query direction: run initialization only in Explore and Dashboards (#100182)

kylebrandt committed:
SQL Expressions: Re-implement feature using go-mysql-server (#99521)

* Under feature flag `sqlExpressions` and is experimental
* Excluded from arm32
* Will not work with the Query Service yet
* Does not have limits in place yet
* Does not working with alerting yet
* Currently requires "prepare time series" Transform for time series viz
 
---------

Co-authored-by: Sam Jewell <sam.jewell@grafana.com>

andresmgot committed:
Advisor: Expose check types as a resource (#100058)

idastambuk committed:
Docs: Add a note on query caching for Cloudwatch datasource (#100180)

eleijonmarck committed:
LBAC for data sources: Update to indicate that we have `experimental` and `GA` (#100187)

update confusing docs

jtvdez committed:
Docs: adding actions_allow_post_url example to plugin docs (#96157)

Proximyst committed:
Advisor: Assert new fields (#100199)

torkelo committed:
Dashboard: Simpify is empty handling (#100189)

* Dashboard: Simpify is empty handling

* remove unused imports

* Update

* Update

* Update

* add code for other layouts

---------

Co-authored-by: Victor Marin <victor.marin@grafana.com>

harisrozajac committed:
Dashboard Schema V2: Introduce __legacyStringValue and deprecate string type for query prop in QueryVariableSpec (#99716)

* Introduce __legacyStringValue and deprecate string type for query

* Fix tests

* Fix tests

* remove default

* kind should default to default ds if variable doesn't have ds field

* lint

* getDefaultDataSourceRef should not return undefined

alyssajoyner committed:
Remove type assertions

nmarrs committed:
Image Renderer: Minor refactor cleanup (#100089)

andresmgot committed:
Drone: Add apps directory for backend tests (#100204)

alyssajoyner committed:
Lint

harisrozajac committed:
Dashboard Schema V2: E2E setup (#99843)

* basic setup

* update CODEOWNERS

* update name

* add temp test that ensures we are loading schema V2 json in the UI

* update language

* test with yarn cache and combine steps

* revert combine

* remove commented out code

* Run current dashboard suite, make workflow optional

* make job always succeed

* Remove temp v2 suite

* don't run on draft PRs

* command for old arch

kalleep committed:
Folder+Dashboard: Resolve parent folders as service in search (#100185)

Resolve parent folders as service to guarantee that we can fetch the title

sarahzinger committed:
Add log line for unexpected queries in ds querier (#100147)

joshhunt committed:
Slider: Fix text input box being too wide (#100138)

* Fix Input width from className not being respected

* Also use width prop in Slider

kalleep committed:
Authz: client cache (#100195)

* Reduce client permissions cache for authz client

* Adjust server cache ttl

JacobsonMT committed:
Alerting: Upgrade grafana/alerting to use EmbeddedContents (#99983)

* Upgrade grafana/alerting to include EmbeddedContents for email images

gassiss committed:
[search] Legacy search fallback support legacy query params (#99765)

* add support for deleted query param

* support tag query param in modes 2 and below

* handle dashboardIds

* hhandle dashboardUIDs

* handle folderUIDs query param

* handle page query param when hitting legacy storage

* handle sort query param

* handle type query param

* re-enable search fallback

* remove folder search workaround and fix /api/search to return both folders and dashboards when no title or type is provided

---------

Co-authored-by: Stephanie Hingtgen <stephanie.hingtgen@grafana.com>

tskarhed committed:
Combobox: Fix list not being virtualized initially in some cases (#100188)

* Combobox: Set arbitrary initial max size

* Remove ?

* Set initial values to 0

mdvictor committed:
DynamicDashboards: Revert unnecessary empty page on new layouts (#100218)

Revert unnecessary empty page on new layouts

alexander-akhmetov committed:
Alerting: Promote alertingSaveStateCompressed flag to public preview (#99935)

samjewell committed:
SQL Expressions: Bump go-mysql-server dependency to `main` (#100222)

Bump go-mysql-server dependency to `main`

This is so we receive a few bug fixes - fixes to these issues:

- https://github.com/dolthub/dolt/issues/8807
- https://github.com/dolthub/dolt/issues/8735
- https://github.com/dolthub/dolt/issues/8724

JohnnyQQQQ committed:
feat(unified-storage): keep tags on reduce of dashboards (#100230)

baldm0mma committed:
Chore: Update grabpl version to v3.1.2 (#100157)

baldm0mma/ update grabpl version

baldm0mma committed:
Chore: Update drone.yml signature (#100236)

baldm0mma/update drone signature

bohandley committed:
Explore metrics: Always check that custom var is present for otel dep env migration (#100233)

suntala committed:
UniStore Big Objects: Fix spec rebuilding (#100183)

* Fix big object spec rebuilding and associated test

---------

Co-authored-by: Jean-Philippe Quémémer <jeanphilippe.quemener@grafana.com>

kylebrandt committed:
SQL Expressions: Add more SQLNodes and funcs to allow list (#100227)

sql_expr: Add more Nodes and funcs to allow list

leventebalogh committed:
Advisor: Enable frontend code generation for the `checktype` kind (#100208)

nit: enabling frontend code generation for the `checktype`

kylebrandt committed:
SQL Expressions: Add str_to_date function and unskip test (#100226)

andresmgot committed:
Advisor: Run check steps in parallel (#100200)

xnyo committed:
Plugins: Add synchronous CDN plugin loader (#99096)

* WIP

* Run plugin validations and validation steps sequentially if feature is off

* Remove dependency between sources.Service and pluginscdn.Service

* lint

* Parallelize validation only if class is CDN

* re-generate feature toggles

* remove waitgroup usage

* PR review: Add loader concurrency limit setting

* re-generate feature toggles

* pr review feedback

* fix const name

* Skip module.js validation for cdn plugins

* do not run validation steps in parallel

* lint

* reduce diff

* re-generate feature toggles

* lint

* pr review feedback

* remove leftover config.PluginManagementCfg from sources.Service

aocenas committed:
QueryLibrary: Move to enterprise (#100133)

bfmatei committed:
Dashboards: Finalize refactoring for dynamic dashboards (#100198)

Proximyst committed:
Provisioning: Integration test that Repository resources work (#100197)

* Provisioning: Set up server in practice

* Provisioning: Integration test that Repository resources work

ifrost committed:
Explore: Fix resizing split view with Loki query editor (#100257)

chaudyg committed:
fix: Ensure we check the user identity for search (#100190)

Ensure we check the user identity for search

Clarity-89 committed:
Card: Use div for description (#100272)

oscarkilhed committed:
Dashboards: Save rows and responsive grid layouts in v2 (#100035)

* save rows and responsive grid layouts in v2

* Add back accidentally removed assert

* adress feedback

* use getGridItemKeyForPanel

* Fix import

* fix another import

* Remove RowGridLayout

toddtreece committed:
Dual Writer: Remove list comparisons in mode 2 & 3 (#100215)

gelicia committed:
Panel edit: Remove use of SheetJS (#100161)

Remove SheetJS

torkelo committed:
Themes: Adds 1 more GrafanaCon theme (wip) (#100228)

* Themes: Gloom theme wip

* Themes: Gloom theme wip

* refactor

* A bit brown maybe?

* Update secondary to also be a bit brownish orangy

* Not super happy

* down a bit

* Progress

* Update

* Update

* orange primary again

Proximyst committed:
Provisioning: Sync API with current feature branch (#100252)

* Provisioning: Jobs: Define repository name field

* Provisioning: Jobs: Separate options per job type

* Provisioning: Define a sanitised settings resource

* Provisioning: Jobs: Define a job summary

* Provisioning: Remove linting

* Provisioning: Update docs for a few fields

* Provisioning: Remove HelloWorld

* Provisioning: Replace Repository with Message in job info

* Provisioning: Remove YAML support

* Provisioning: Remove custom folder specification

* Provisioning: Support read-only repositories

* Provisioning: Remove edit options

* Provisioning: Add sync options for repositories

* Provisioning: Add resource statistics

* Provisioning: Make slices atomic lists

* Provisioning: Message list needs to exist even if empty

If we don't do this, we can't clear the messages field, leading to buggy UX.

* Provisioning: Support incremental syncing

* Provisioning: Remove the 'items' subresource workaround

* Provisioning: Add resource list

* Provisioning: Reformat

* Provisioning: Declare new types

* OpenAPI: Generate openapi JSON spec from generated code

* Codegen: Generate OpenAPI spec

* Provisioning: Support generating frontend API

* Codegen: Generate Go code

* Provisioning: Define the base API

* Codegen: Generate frontend endpoints for provisioning

* Refactor: yarn prettier:write

* Provisioning: Tiger team takes ownership

* Chore: Remove dir we haven't added yet

* Provisioning: Remove frontend

* Test: Update example repositories

Clarity-89 committed:
Card: Render description element based on children type (#100276)

bfmatei committed:
Dashboard: Fix for overwriting an edited dashboard in the old architecture (#100247)

AgnesToulet committed:
E2E: Add PDF/CSV/Image comparison tasks (#99104)

* E2E: Add tasks for new reporting tests

* try to switch to pdf-parse

* fix pdf comparison

* add log

* increase threshold

* clean up

* apply review feedback

* improve logs + fix local setup for reporting

* format

andresmgot committed:
Advisor: Replace Error naming with Failure (#100262)

alyssajoyner committed:
Refactor query parser

bfmatei committed:
Dashboards: Add tabs layout (#100127)

alyssajoyner committed:
Merge branch 'main' into alyssa/azure-logs-simple-mode-data-transforms
